Adds a script to run sensitivities (#44)
ucsusa/pypsa-illinois: sensitivity.md

# Variables to change for sensitivities

- **Coal and methane prices.** Fuel prices are modeled as 12 values from a historical year, derived from this PUDL database: https://data.catalyst.coop/pudl/core_eia923__monthly_fuel_receipts_costs.

> - **ATB cases.** The model uses the 2022 ATB, which has three scenarios: "Moderate", "Conservative", and "Advanced". For sensitivity analysis we can ignore "Moderate" and just use "Conservative" and "Advanced".

Reviewer: ```suggestion
- **ATB cases.** The model uses the 2022 ATB, which has three scenarios: "Moderate", "Conservative", and "Advanced". We only use the "Moderate" case, here.
```

| Parameter   | Variable in `config.yml`| Scripts                                       | Values                           |
|-------------|-------------------------|-----------------------------------------------|----------------------------------|
|Fuel prices  |`fuel_cost_year`         |`retrieve_fuel_prices.py`, `add_electricity.py`|2018, 2019, 2020, 2021, 2022, 2023|
|ATB cases    |`scenario`               |`retrieve_costs.py`                            |"Conservative", "Advanced"        |
|Demand growth|`load_growth`            |`add_electricity.py`                           |0.00, 1.00                        |
|Exports      |`total_demand`           |`add_electricity.py`                           |136e6, 185e6                      |

Total possibilities = 6x2x2x2 = 48

The order in the snakemake workflow of the rules we need to modify:

| Rule | Script                  |
|------|-------------------------|
|3     |`retrieve_costs.py`      |
|4     |`retrieve_fuel_prices.py`|
|11    |`add_electricity.py`     |

# Remove ATB cases

> There are no conservative or advanced cases for nuclear in the ATB which causes the model to crash. The simplest workaround is just to not do sensitivites on ATB scenarios.

Reviewer: ```suggestion
Since the NREL ATB does not calculate all scenarios (conservative, moderate, advanced) for all technologies, we did not include a sensitivity around capital cost in the model.
```
I think saying something caused the model to crash is a bit misleading. The model didn't crash -- it didn't even run! The workflow tried to automatically access data that doesn't exist -- because NREL didn't provide it in the ATB.

Author: Good point, though I still think we should be clear that the model won't run if you set those ATB scenarios (rather than us just not including them). There are data for the other technologies which may be of interest in a sensitivity.
